avm/res/databricks/workspace/main.bicep

```diff
@@ -717,3 +721,11 @@ type diagnosticSettingType = {
   @description('Optional. The full ARM resource ID of the Marketplace resource to which you would like to send Diagnostic Logs.')
   marketplacePartnerResourceId: string?
 }[]?
+
+type defaultCatalogType = {
+  @description('Optional. Set the name of the Catalog.')
+  initialName: '' // This value cannot be set to a custom value. Reason --> 'InvalidInitialCatalogName' message: 'Currently custom initial catalog name is not supported. This capability will be added in future.'
```

Comment:
This would be a great solution - if - types would allow for default values. Unfortunately they don't, and every value that you provide after the `:` just acts as an 'allowedValue'. As such the property would still be suggested to the user, but they could only set this single value. 

Unless that's intended, I'd give the below a shot

```bicep 
resource workspace 'Microsoft.Databricks/workspaces@2024-05-01' = {
 (...)
 properties: {
    defaultCatalog: !empty(defaultCatalog) ? {
      initialName: ''
      initialType: defaultCatalog.?initialType
    } : null
 (...)
 } 
}

type defaultCatalogType = {
  // This value cannot be set to a custom value. Reason --> 'InvalidInitialCatalogName' message: 'Currently custom initial catalog name is not supported. This capability will be added in future.'
  // @description('Optional. Set the name of the Catalog.')
  // initialName: string

  @description('Required. Choose between HiveMetastore or UnityCatalog.')
  initialType: 'HiveMetastore' | 'UnityCatalog'
}

Reply:
In the test case you can then just provide the parameter as
```bicep
defaultCatalog: {
  initialType: 'UnityCatalog'
}
```

Reply:
Ok, thanks, but what about the Default deployment and WAF, we don't want to set defaultCatalog unles we have to, but if I leave it out the main.test.bicep do you think it will work. I'll give it a try now

Reply:
oh, but I think the if is emtpy bit will do that for us. never mind, ignore my above comment